File: samples/java/demo/app/DemoModule.java
import com.google.inject.AbstractModule;
import securesocial.plugin.RuntimeEnvironment;
import service.MyEnvironment;


public class DemoModule extends AbstractModule {
    @Override
    protected void configure() {
        MyEnvironment environment = new MyEnvironment();
        bind(RuntimeEnvironment.class).toInstance(environment);
    }
}

File: samples/scala/demo/app/DemoModule.scala
import com.google.inject.{ TypeLiteral, Scopes, AbstractModule }
import net.codingwell.scalaguice.ScalaModule
import securesocial.plugin.{ BasicProfile, RuntimeEnvironment }
import service.{ MyEnvironment, DemoUser }

class DemoModule extends AbstractModule with ScalaModule {
  override def configure() {
    val environment: MyEnvironment = new MyEnvironment
    bind(new TypeLiteral[RuntimeEnvironment] {}).toInstance(environment)

  }
}
